simd/unit_tests/include/SIMDTesting_Utilities.hpp

```diff
@@ -53,12 +59,14 @@ inline void host_check_equality(
   }
   using mask_type =
       typename Kokkos::Experimental::basic_simd<T, Abi>::mask_type;
-  if constexpr (std::is_same_v<Abi, Kokkos::Experimental::simd_abi::scalar>) {
-    mask_type mask(KOKKOS_LAMBDA(std::size_t i) { return (i < nlanes); });
-    checker.equality((expected_result == computed_result) && mask, mask);
-  } else {
-    mask_type mask([=](std::size_t i) { return (i < nlanes); });
-    checker.equality((expected_result == computed_result) && mask, mask);
+  if constexpr (std::is_integral_v<T>) {
```

Comment:
Why do you restrict the test to integral type? The PR description only mentions the change in `equality()`.

Reply:
It's because of `expected_result == computed_result`. The expression fails for floating point types with icpc, because the simd equality operators aren't set up to check with tolerances.
But this restriction should be limited to icpc builds. Fixed.